### ISSUE TITLE
mask not reversed for RNN?

### ISSUE THREAD

**Issue by strin** [fchollet/keras#3737]
In class `keras.core.layers.recurrent`:

``` python
def compute_mask(self, input, mask):
        if self.return_sequences:
            return mask
        else:
            return None
```

But if `go_backwards=True`, the mask should be reversed in time.
